art/attacks/inference/membership_inference/label_only_boundary_distance.py

```diff
@@ -196,6 +202,62 @@ def calibrate_distance_threshold(
 
         self.distance_threshold_tau = distance_threshold_tau
 
+    def calibrate_distance_threshold_unsupervised(self, top_t: int, num_samples: int, max_queries: int, **kwargs):
+        """
+        Calibrate distance threshold on randomly generated samples, choosing the top-t percentile of the noise needed
+        to change the classifier's initial prediction.
+
+        | Paper link: https://arxiv.org/abs/2007.15528
+
+        :param top_t: Top-t percentile.
+        :param num_samples: Number of random samples to generate.
+        :param max_queries: Maximum number of queries.
+        :Keyword Arguments for HopSkipJump:
+            * *norm*: Order of the norm. Possible values: "inf", np.inf or 2.
+            * *max_iter*: Maximum number of iterations.
+            * *max_eval*: Maximum number of evaluations for estimating gradient.
+            * *init_eval*: Initial number of evaluations for estimating gradient.
+            * *init_size*: Maximum number of trials for initial generation of adversarial examples.
+            * *verbose*: Show progress bars.
+        """
+        from art.attacks.evasion.hop_skip_jump import HopSkipJump
+
+        x_min, x_max = self.estimator.clip_values
```

Comment:
Need some solution for the case that estimator.clip_values is not set (it is an optional param when initializing an estimator).

Reply:
If there was an input array `x` passed to the function I would do a similar approach to the hsj implementation (but there isn't :crying_cat_face: ):
```python
# Get clip_min and clip_max from the classifier or infer them from data
if self.estimator.clip_values is not None:
    clip_min, clip_max = self.estimator.clip_values
else:
    clip_min, clip_max = np.min(x), np.max(x)
```
Another solution would be to do a check at the beginning of the function and throw an error if clip_values is not set.

Or maybe the best solution would be to do a check, and if it's None then set default values 0 and 1 and throw a warning. Something like this:
```python
if self.estimator.clip_values is not None:
    x_min, x_max = self.estimator.clip_values
else:
    x_min, x_max = 0, 1
    raise RuntimeWarning("The estimator has no clip_values set. Generated feature values will range between 0 and 1.")
```

Reply:
Will the calibration work properly even if wrong (default) min and max values are used? Or is this critical for the correctness of the threshold? If it's the latter, I prefer to raise an exception instead of using default values so that the user knows that these must be set. @beat-buesser what are your thoughts on this?

Reply:
I changed it to raise a `RuntimeError` in the newest commit ([a0e924c](https://github.com/Trusted-AI/adversarial-robustness-toolbox/pull/1197/commits/a0e924c751a4ee47f34e5b06a3aef66c07651772)). Is this ok @beat-buesser ?